portal/static/js/main.js

```diff
@@ -2118,60 +2118,93 @@ var assembleContent = {
             var siteId = siteIdField.val();
 
 
-            if (hasStudyId || hasSiteId) {
-                var identifiers = null;
-                //get current identifier(s)
-                $.ajax ({
-                    type: "GET",
-                    url: "/api/demographics/"+userId,
-                    async: false
-                }).done(function(data) {
-                    if (data && data.identifier) {
-                        identifiers = [];
-                        (data.identifier).forEach(function(identifier) {
-                            if (identifier.system != SYSTEM_IDENTIFIER_ENUM["external_study_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["external_site_id"] &&
-                                identifier.system != SYSTEM_IDENTIFIER_ENUM["practice_region"]) {
-                                identifiers.push(identifier);
-                            }
-                        });
-                    }
-                }).fail(function(xhr) {
-                   tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
-                });
+            var identifiers = null;
+            /*
+             * get current identifier(s)
+             */
+            $.ajax ({
+                type: "GET",
+                url: "/api/demographics/"+userId,
+                async: false
+            }).done(function(data) {
+                if (data && data.identifier) {
+                    identifiers = [];
+                    (data.identifier).forEach(function(identifier) {
+                        identifiers.push(identifier);
+                    });
+                }
+            }).fail(function(xhr) {
+               tnthAjax.reportError(userId, "api/demographics"+userId, xhr.responseText);
+            });
 
+            /*
+             * NOTE: this will save study Id or site Id only if each has a value
+             * otherwise if each is empty, it will be purged from the identifiers
+             */
+
+            if (hasStudyId || hasSiteId) {
```

Comment:
Is this condition still necessary? You check `hasStudyId` and `hasSiteId` individually inside of it (I may have misread the indentation level though...)

Reply:
@ivan-c ivan, renamed the variables so it is more clear - study Id field only exists in eproms wherease site id field only exists in truenth - so need to check each field's presence before saving the value for each